Lib/test/test_functools.py

```diff
@@ -2792,6 +2792,41 @@ def _(arg: int | float):
         self.assertEqual(f(1), "types.UnionType")
         self.assertEqual(f(1.0), "types.UnionType")
 
+    def test_union_None(self):
+        @functools.singledispatch
+        def f(arg):
+            return "default"
+
+        @f.register
+        def _(arg: typing.Union[str, None]):
+            return "typing.Union"
+
+        @f.register
+        def _(arg: int):
+            return "types.UnionType"
```

Comment:
No, this does not test `types.UnionType`. Remove this registration, it is not needed.